src/views/AddItem.jsx

```diff
@@ -1,3 +1,83 @@
-export function AddItem() {
-	return <p>Hello from the <code>/add-item</code> page!</p>
+import { React, useState } from 'react';
+
+import { addItem } from '../api/firebase';
+
+//we need listToken (listID) passing from App.js
+
+export function AddItem({ listToken }) {
+	const [itemName, setItemName] = useState('');
+	const [daysUntilNextPurchase, setDaysUntilNextPurchase] = useState(7);
```

Comment:
We have a default daysUntilNextPurchase values of 7, because it is the default radio button selected. But we need to also hard code the value of 7 on the form submit function. Any ideas on how to link the value of daysUntilNextPurchase to the radio button with defaultChecked

Reply:
I couldn't figure out another way to do this

Reply:
so I feel the most economic way to make a radio button in React that is bound to state is to use `checked` rather than `defaultChecked`. We set our 'default' by setting our initial state value.
```
<input
    type="radio"
    id="7"
    name="itemFrequencyInDays"
    value="7"
    checked={daysUntilNextPurchase === 7}
    onChange={handleFrequencyInput}
/>
```
this has the added benefit of making sure one of them will always be selected, so we can remove required from the radio buttons

one thing to look out for on your eventHandler (if you make the above change) is make sure you are converting `e.target.value` to a number since the radio button will turn your `7`s into `"7"`s

`const handleFrequencyInput = (e) => setDaysUntilNextPurchase(+e.target.value);`

Reply:
🙌 thank you for this 🔥 tip!